### PR TITLE
Project Mongo API - Julia Ros

### PR DESCRIPTION
**Pull request by julialeonieros**
https://books-mongodb.herokuapp.com/

### CODE REVIEW

**Review by CarolineF3 — CHANGES_REQUESTED** [review]
Really nice project, well done!✨🎉 It's great that you're using try and catch. All of the endpoints works great except for top10, not sure why.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-## The problem
-
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+## Endpoints
+* '/' - See a list of endpoints
+* '/book' - See all books
+* '/books?author=[...]&title=[...] - Search for authors and/or title
+* '/books/[id] - Search for book by ID
+* '/books/isbn/[isbn] - Search for book by ISBN
+* '/books/top10' - Get a list of the 10 books with the highest rating
```
Comment (KarolCislowski) [inline README.md:21]: Nice documentation 😸

File: server.js [hunk 2]
```diff
@@ -1,38 +1,122 @@
...
+  const { author, title } = req.query
+  const authorRegex = new RegExp(author, 'i')
+  const titleRegex = new RegExp(title, 'i')
+
+  try {
+    const books = await Book.find({
+      authors: authorRegex,
+      title: titleRegex
+    })
+    res.json(books)
```
Comment (KarolCislowski) [inline server.js:77]: nice filtering 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,38 +1,122 @@
...
+      res.status(404).json('Sorry, no book with that ISBN')
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid ISBN', details: error })
+  }
+})
+
+// endpoint to get a list with the 10 books that has the highest rating
+app.get('/top10', async (req, res) => {
+  const top10 = await Book.find().sort({ average_rating: -1 }).limit(20).exec()
```
Comment (KarolCislowski) [inline server.js:116]: top 10 and limit 20? 😜

**Review by KarolCislowski — APPROVED** [review]
Great API 😄

**Pull request closed without merging**